test/CMakeLists.txt

```diff
@@ -19,8 +19,18 @@ include_directories(source)
 #-------------------------------------------------------------------------------------------
 set(CMAKE_CXX_STANDARD 11)
 
-if(UNIX)
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+	SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+endif()
+
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+	set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -D_DEBUG")
+	set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -D_DEBUG")
+endif()
+
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+	# Disable some (maybe useless) warnings
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-shift-negative-value -Wno-pointer-bool-conversion")
```

Comment:
maybe mark this as a todo.  I'd like to enable the negative shift warning... so I'm curious what code is triggering that.